### PR TITLE
Jwoo/toggle view update

### LINKED ISSUE #3978: [Ios] [Adaptive Cards Input Toggle(CheckBox) display issue in Iphone]
# Platform

What platform is your issue or question related to? (Delete other platforms).

* iOS

# Author or host

Ms Teams

# Details

We have created bot using Microsoft bot-framework which is integrated to Microsoft teams. We have created few adaptive cards for interactive conversation with bot. One of the Adaptive card is causing design issue in IPhone X Microsoft Teams Application.

We used Adaptive card template for building this card which displays the list of Tasks along with status of the task. Each task has Input.Toggle (Checkbox) and at the end a button will be displayed. When Checkbox's are selected and button is clicked. Sample design as shown [Adaptive Card Design Preview](https://i.stack.imgur.com/CQ1tp.png)

We are sending new card with task details where they can edit the status and so on. Please find the JSON data below.

Card Payload Editor Json Data:
[Card Payload.txt](https://github.com/microsoft/AdaptiveCards/files/4572961/Card.Payload.txt)

Sample Data Editor:
[Sample Data.txt](https://github.com/microsoft/AdaptiveCards/files/4572965/Sample.Data.txt)

The InputToggle is inside the column set with **width** as **auto** and it is perfectly rendering in [Microsoft Teams Desktop](https://i.stack.imgur.com/3ahYn.png) and  [ Android App ](https://i.stack.imgur.com/Opo9i.jpg) but whereas in [IOS app](https://i.stack.imgur.com/S9MLP.png) it is not showing.

When we set the **width** property to **stretch**, InputToggle is getting displayed in [IOS App](https://i.stack.imgur.com/hHGaH.png) in big size but in [Desktop App](https://i.stack.imgur.com/ojihc.png) and [Android App](https://i.stack.imgur.com/VvSxF.png) the column is pushing and occupying much space. 

Please help me so that we can display the Input.Toggle in IOS Microsoft Teams without breaking the design in Desktop and Android Teams App.

Adding final JSON sent to the bot. The columns in the column set will be changed dynamically based on the ask list in database.

[Final Json.txt](https://github.com/microsoft/AdaptiveCards/files/4572957/Final.Json.txt)

Thank you.

### LINKED ISSUE #4052: [iOS][ColumnSet] [Auto and stretch columns have the same behaviour]
# Platform

What platform is your issue or question related to? (Delete other platforms).

* iOS

# Author or host

Host - Teams iOS

# Version of SDK

latest master

# Details

ColumnSet does not stretch the column of type 'stretch' when another column has type 'auto'.
The behaviour on Desktop and Android seems to be the expected behaviour.
Attaching screenshots and card JSON:
**iOS:**
iOS input Toggle
**Desktop:**
Desktop input Toggle

**Card JSON:**
[inputToggleColumns.txt](https://github.com/microsoft/AdaptiveCards/files/4680794/inputToggleColumns.txt)

### PR DESCRIPTION
**Pull request by jwoo-msft**
## Related Issue
Fixed #4052, Fixed #3978 

## Description
#4052 mentioned that "stretch" and "auto" did not work very well when Input.Toggle is the only item of Column. This is an expected behavior since the input toggle is implemented using UITableView, and label space provided by UITableView has string compression resistance. However, using UITableView as a superview for Input.Toggle is too heavy since UITableView has functions that can't be utilized in Input.Toggle. Implemented Input.Toggle as a XIB instance that allows easier customization of Input.Toggle, and used simpler UI inside the XIB. 

## How Verified
How you verified the fix, including one or all of the following: 
1. attached screen shots.
**AFTER**
image

**BEFORE**
image

2. existing json cards are verified.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/4204)

### CODE REVIEW

**Comment by shalinijoshi19** [comment]
@golddove is added to the review. #Closed

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRInputToggleRenderer.mm [hunk 1]
```diff
@@ -37,38 +38,47 @@ - (UIView *)render:(UIView<ACRIContentHoldingView> *)viewGroup
 {
     std::shared_ptr<HostConfig> config = [acoConfig getHostConfig];
     std::shared_ptr<BaseCardElement> elem = [acoElem element];
-    std::shared_ptr<ToggleInput> toggleBlck = std::dynamic_pointer_cast<ToggleInput>(elem);
+    std::shared_ptr<ToggleInput> adaptiveToggleInput = std::dynamic_pointer_cast<ToggleInput>(elem);
 
     NSBundle *bundle = [NSBundle bundleWithIdentifier:@"MSFT.AdaptiveCards"];
     if (!bundle) { // https://github.com/Microsoft/AdaptiveCards/issues/1834
```
Comment (shalinijoshi19) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRInputToggleRenderer.mm:44]: >Microsoft [](start = 41, length = 9)

Nit: This looks to be a TODO; In that case, can you update teh comment to be more explicit following a "TODO: [Link to the tracking issue]  Log warnings and errors"

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRInputToggleRenderer.mm [hunk 2]
```diff
@@ -37,38 +38,47 @@ - (UIView *)render:(UIView<ACRIContentHoldingView> *)viewGroup
...
-    inputTableView.delegate = dataSource;
-    inputTableView.dataSource = dataSource;
 
+    ACRToggleInputView *toggleView = [[ACRToggleInputView alloc] initWithFrame:CGRectMake(0, 0, viewGroup.frame.size.width, viewGroup.frame.size.height)];
+
+    toggleView.title.text = [NSString stringWithCString:adaptiveToggleInput->GetTitle().c_str() encoding:NSUTF8StringEncoding];
+    toggleView.title.textColor = getForegroundUIColorFromAdaptiveAttribute(config, viewGroup.style);
+    toggleView.title.adjustsFontSizeToFitWidth = NO;
+    if (!adaptiveToggleInput->GetWrap()) {
+        toggleView.title.numberOfLines = 1;
```
Comment (shalinijoshi19) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRInputToggleRenderer.mm:54]: >1; [](start = 41, length = 2)

In general for code readability avoid hardcoded literals and magic numbers; Trying to understand the logic here - So if wrap property isnt specified we interpret/default to 1 for its title line count? This almost feels like an initialization operation that say can be handled at the constructor for ACRToggleInputView no?

Comment (jwoo-msft) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRInputToggleRenderer.mm:54]: iOS renderer defaults number of lines to zero. If zero, text box will keep increasing its height with longer text. iOS renderer limits number of lines to 1, and its effect is same as having wrap property set to "false".

**Pull request merged**